### ISSUE TITLE
Long press support on mobile app (iPad/iPhone/android) like obsidian does

### ISSUE THREAD

**Issue by STMRabbit** [siyuan-note/siyuan#9096]
### In what scenarios do you need this feature?

This made these things not possible on mobile: 
1. moving icons on panels is not possible on iPad.
2. "more" menu (right click menu) is hard to touch (because menu button is small on mobile devices)
3. a useless "text selecting" calling on some elements is useless and disturbing

![IMG_0905](https://github.com/siyuan-note/siyuan/assets/112855065/6d9aaf40-25fa-45d7-a4c8-bf6c4b298e79)

For now, long press elements on mobile devices is calling "select text", which doesn't make sense, it should be like:
1. when long press text editor area: select text
5. when long press title **edit box** (not titles in documents tree) : select text
6. when long press other elements (include titles in documents tree, buttons etc.) : pop out right click menu ("more" menu).

### Describe the optimal solution

Maybe long press also should be supported on desktop app as well, considering there are desktop devices which support touch screen like Microsoft Surface or Pinetab. But I don't have those devices, so I can't give a clear suggestion.

### Describe the candidate solution

At least disable long press calling on non-editor elements because that feature is useless and even disturbing. 

### Other information

mobile note taking app with same tec stack (for reference)
joplin   
obsidian

**Comment by Vanessa219** [comment]
1. iPad issue: https://github.com/siyuan-note/siyuan/issues/4238
2. What function does the "More" menu (right-click menu) refer to?
3. Disable select left top bar will avoid this issue

There is no context menu on the iPhone, so we will disable the select event

**Comment by STMRabbit** [comment]
Hello, thanks for fixing that! And here are the answers referring to your question:  

# I

> iPad issue: https://github.com/siyuan-note/siyuan/issues/4238

i apologize for my expressing, but what I was referring is the issue that: I can't move these buttons (shows in the image below) on iPad, because moving these buttons need to right-click the **mouse** to call the `context menu`, which iPad doesn't usually have.

![image](https://github.com/siyuan-note/siyuan/assets/112855065/0f3a5e07-53a8-42ca-9759-16bb7d22712a)

and i was just saying maybe long press to call the context menu (AKA "more" menu or "right-click" menu is a designing option)

# II

>What function does the "More" menu (right-click menu) refer to?    

i meant this button:

![image](https://github.com/siyuan-note/siyuan/assets/112855065/88dac57e-d32b-445e-b609-f4b77f8522a1)


on iOS and android, when long press a note or notebook in `doc tree`, it pops out `text selecting cursor` which doesn't make sense, it should be equal to when you press the "menu" button, because it doesn't make sense to select text at that field (maybe you already fixed it in last commit?).   
in another word: long press a note or notebook in `doc tree` should pop out the context menu (AKA "more" menu or "right-click" menu) , instead of calling text-selecting cursor.


thank you again!

**Comment by Vanessa219** [comment]
* [Right-click can not open the menu on the dock icon](https://github.com/siyuan-note/siyuan/issues/4238)
* [Long press the file tree item to open the right menu on the mobile](https://github.com/siyuan-note/siyuan/issues/9111)
* Open about -> workspace -> open -> click select element

**Comment by STMRabbit** [comment]
thank you again! 

# I

>Right-click can not open the menu on the dock icon    

yes thank you! and just to be more clear, what I meant is that since iPad doesn't have a mouse, so all the operate which need  right-clicking the mouse, is not able to do on iPad.  

# II

>Long press the file tree item to open the right menu on the mobile      
  
yes, indeed, thank you so much!  

# III

>Open about -> workspace -> open -> click select element  

i apologize for my misunderstood, is this a TODO or an instruction? cuz if it's an instruction, i can't find any element/button related to " workspace" or similar thing in about `page`.  And if it's a TODO, do you need me to open another issue to describe it? 

Thank you!

**Comment by Vanessa219** [comment]
Sorry, I was talking about mobile, if it's an iPad, try this:

<img width="332" alt="image" src="https://github.com/siyuan-note/siyuan/assets/970828/88d413ae-9292-44de-aed4-11aee9417a83">

**Comment by STMRabbit** [comment]
yes, thank you for your helping! but it seems as i said, that button in your screenshot can't delete a profile/workspace either. On iPad i can only remove a workspace/profile from the _list_ by touch the "X" icon, the entire files hadn't been deleted, still exist, because I can open it later again by "open" button even after I removed it. 
  
Also as you mentioned, i tested the iPhone client too, and it turns out I can't completely delete a workspace on iPhone  either. (the "-" button just remove it from list, too)

on android, i can delete a workspace using root permission +  file manager, but can't delete it within siyuan app itself.

sorry if it's just because i didn't find it, but it's at least not that obvious.

**Comment by Vanessa219** [comment]
It used to be physically deleted, but some users often deleted it by mistake, so it was later changed to remove it from the workspace list.

**Comment by STMRabbit** [comment]
Thank you for your answer! Maybe it's a nice design, however personally I think it's also important to be able to physically delete a workspace (maybe it could be located deeply in settings or something to prevent accident), since on iOS devices, the user can't really use file manager to manage files.   
But don't worry, it's just a suggestion, user can still uninstall siyuan and install again for that purpose, if you believe it's better.

**Comment by 88250** [comment]
@STMRabbit We will implement it in the next version, thanks  https://github.com/siyuan-note/siyuan/issues/9134

**Comment by STMRabbit** [comment]
Thank you so much for the quick fix! In case I expressed myself incorrectly last time, the term "uninstall SiYuan and install it again" was meant to describe the procedure for clearing data. It's not intended as a threat, like "if you don't implement it, the user will uninstall and leave." :) 
I love SiYuan; it's the best app I've ever used. Thank you so much!
